ambition_utils/rrule/models.py

```diff
@@ -295,6 +307,22 @@ def convert_to_utc(self, dt):
 
         return dt
 
+    def offset(self, dt, reverse=False) -> datetime:
+        """
+        Offsets a given datetime by the number of days specified by day_offset.
+        :param dt: The datetime to offset by day_offset
+        :param reverse: Reverse the offset calculation.
+        :return dt:
+        """
+        # The offset gets multiplied by 1 or -1 depending on offset direction
+        multiplier = -1 if reverse else 1
```

Comment:
instead of having a reverse flag why don't we just store negative or positive integers? You have a reverse flag but the system would have no way of knowing if it should be reversed or not and we are assuming true. By storing a positive or negative integer it would allow both directions

Reply:
It does support negative and positive integers. This flag is only for `get_next_occurrence()` which needs to convert the `last_occurrence` _back_ in order to pass `rule_set.after()` a date it is expecting.